R/comparePlot.R

```diff
@@ -141,133 +233,121 @@ comparePlot <- function(compareDF,
     }
 
-    # plotType
-    if (plotType == "canvasXpress") {
-        # adding alpha to colors
-        symbolFill <- sapply(symbolFill, rgbaConversion, alpha = alpha, USE.NAMES = FALSE)
+    y_range <- compareDF %>%
+        dplyr::select(ylabel) %>%
+        dplyr::summarise(across(everything(), list(min, max))) %>%
+        dplyr::rename_with(~ c("min", "max"))
+    if (plotType == "canvasxpress") {
+        # adding transparency to colors
+        symbolFill <- sapply(symbolFill, rgbaConversion, alpha = transparency, USE.NAMES = FALSE)
         decorations <- list()
-
         if (!is.null(crosshair)) {
-            crosshair <- rgbaConversion(crosshair, alpha = alpha)
-            decorations <- getCxPlotDecorations(decorations, color = crosshair, width = refLineThickness, x = 0)
-            decorations <- getCxPlotDecorations(decorations, color = crosshair, width = refLineThickness, y = 0)
+            decorations <- getCxPlotDecorations(decorations,
+                                                color = rgbaConversion(crosshair, alpha = transparency),
+                                                width = refLineThickness,
+                                                x     = 0)
         }
-
         if (!is.null(referenceLine)) {
-            referenceLine <- rgbaConversion(referenceLine, alpha = alpha)
-            y_range <- range(subset(compareDF, select = c(ylabel), drop = TRUE), na.rm = TRUE)
             decorations <- getCxPlotDecorations(decorations = decorations,
-                                                color       = referenceLine,
+                                                color       = rgbaConversion(referenceLine, alpha = transparency),
                                                 width       = refLineThickness,
-                                                x           = ceiling(y_range[1]),
-                                                y           = floor(y_range[2]))
+                                                x           = ceiling(y_range$max),
+                                                y           = floor(y_range$min))
         }
-
-        if (missing(footnote)) {
-            footnote <- NULL
-        }
-
-        cx.data <- subset(compareDF, select = c(xlabel, ylabel))
-        colnames(cx.data) <- c(xlab, ylab)
+        cx.data <- compareDF %>%
+            dplyr::select(c(xlabel, ylabel)) %>%
+            dplyr::rename_with(~ c(xlab, ylab))
         if (sigMeasurePlot) {
-            cx.data <- round(cx.data, digits = 2)
-
-            var.annot <- subset(compareDF, select = c(group), drop = FALSE)
-            colnames(var.annot) <- "Group"
+            cx.data   <- round(cx.data, digits = 2)
+            var.annot <- compareDF %>%
+                dplyr::select(group) %>%
+                dplyr::rename_with(~ c("Group"))
             colorBy <- "Group"
             sizeBy  <- "Group"
             colors  <- symbolFill[c(1,4,2,3)]
```

Comment:
Connie had mentioned in one of the previous PRs that we are to use symbolColor instead of symbolFill for canvasXpress

Reply:
for both plots?

Reply:
For all the plots .. Symbol fill can be taken out and symbol color will be used as for both fill and outline.

Reply:
done